framework/src/geomsearch/FindContactPoint.C

```diff
@@ -222,7 +222,8 @@ findContactPoint(PenetrationInfo & p_info,
   else
   {
     p_info._normal = RealGradient(dxyz_dxi[0](1),-dxyz_dxi[0](0));
-    p_info._normal /= p_info._normal.size();
+    if (std::fabs(p_info._normal.size()) > 1e-15)
```

Comment:
Fairly tight tolerance... I guess if this works for you, then it's fine with me.